CONTRIBUTING.md

```diff
@@ -63,11 +63,33 @@ We require that your commit messages match our template. The easiest way to do t
 
     git config commit.template dev/git.commit.template
 
+The default policy for taking contributions is “Squash and Merge” - because of this, the commit message format rule above, applies to the PR rather than every commit contained within it. 
+
 ### Make sure Tests work on Linux
 
-SwiftNIO uses XCTest to run tests on both macOS and Linux. While the macOS version of XCTest is able to use the Objective-C runtime to discover tests at execution time, the Linux version is not. 
+SwiftNIO uses XCTest to run tests on both macOS and Linux. While the macOS version of XCTest is able to use the Objective-C runtime to discover tests at execution time, the Linux version is not (prior to swift 5.1). 
 For this reason, whenever you add new tests **you have to run a script** that generates the hooks needed to run those tests on Linux, or our CI will complain that the tests are not all present on Linux. To do this, merely execute `ruby ./scripts/generate_linux_tests.rb` at the root of the package and check the changes it made.
 
+### Make sure your patch works for all supported versions of swift
+
+The CI will do this for you.  You can use the docker-compose files included if you wish to check locally.  Currently all versions of swift >= 5.0 are supported.
+
+### Make sure your code is performant
+
+SwiftNIO has been created to be high performance.  The integration tests cover some measures of performance including allocations which should be avoided if possible.
+
+### Formatting
+
+Try to keep your lines less than 120 characters long so github can correctly display your changes.
+
+It is intended SwiftNIO will use the swift-format tool in the future to bring consistency to code formatting.
```

Comment:
I don't think we need this.

Reply:
Hmm, don't think we strictly speaking _need_ it but I think it's what we're intending once the discussion is resolved so I'd be in favour of mentioning it.

It may be worth adding at what point we're intending to do this which is I think once there's an official style guide and the Swift stdlib itself has been reformatted. I think we should then follow suit and reformat too.

Reply:
The evo proposal to link to is SE-250: https://github.com/apple/swift-evolution/blob/master/proposals/0250-swift-style-guide-and-formatter.md

Reply:
Link to SE-250 added.